core/src/debug_ui/avm1.rs

```diff
@@ -39,23 +47,273 @@ impl Avm1ObjectWindow {
+                    re.request_focus();
+                    self.focus_text_edit = false;
+                }
+                match self.value_edit_buf.parse::<f64>() {
+                    Ok(num) => {
+                        if ui.input(|inp| inp.key_pressed(egui::Key::Enter))
+                            || ui.set_button().clicked()
+                        {
+                            new_val = Some(num);
+                            self.edited_key = None;
+                        }
+                    }
+                    Err(e) => {
+                        ui.add_enabled(false, egui::Button::new(CHECKMARK_ICON))
+                            .on_disabled_hover_text(e.to_string());
+                    }
+                }
+                if ui.cancel_button().clicked() {
+                    self.edited_key = None;
+                }
             });
-        keep_open
+        } else {
+            ui.horizontal(|ui| {
+                let num_str = num.to_string();
+                ui.label(&num_str);
+                if ui.edit_button().clicked() {
+                    self.edited_key = Some(key.as_wstr().to_owned());
+                    self.value_edit_buf = num_str;
+                    self.focus_text_edit = true;
+                }
+            });
+        }
+        new_val
+    }
+    fn string_edit_ui(
+        &mut self,
+        ui: &mut Ui,
+        key: &AvmString,
+        string: AvmString,
+    ) -> Option<String> {
+        let mut new_val = None;
+        ui.horizontal(|ui| {
+            if self
+                .edited_key
+                .as_ref()
+                .is_some_and(|edit_key| *edit_key == key.as_wstr())
+            {
+                let re = ui.add(TextEdit::singleline(&mut self.value_edit_buf).desired_width(96.0));
+                if self.focus_text_edit {
+                    re.request_focus();
+                    self.focus_text_edit = false;
+                }
+                if ui.set_button().clicked() {
+                    new_val = Some(self.value_edit_buf.take());
+                    self.edited_key = None;
+                }
+                if ui.cancel_button().clicked() {
+                    self.edited_key = None;
+                }
+            } else {
+                ui.label(string.to_utf8_lossy());
+                if ui.edit_button().clicked() {
+                    self.value_edit_buf = string.to_string();
+                    self.edited_key = Some(key.as_wstr().to_owned());
+                    self.focus_text_edit = true;
+                }
+            }
+        });
+        new_val
+    }
+}
+
+/// Dropdown menu indicating the type of the value, as well as letting the
+/// user set a new type.
+fn show_value_type_combo_box<'gc>(
+    ui: &mut Ui,
+    key: &WStr,
+    value: &Value<'gc>,
+    mutation: &Mutation<'gc>,
+) -> Option<Value<'gc>> {
+    let mut new = None;
+    egui::ComboBox::new(egui::Id::new("value_combo").with(key), "Type")
```

Comment:
Perhaps the "Type" label is a bit redundant, and the dropdown menu is obvious enough without the label.